scripts/release.promote.sh

```diff
@@ -9,8 +9,13 @@ echo "Loading release tool"
 "$dir/goinstall.sh" "github.com/keybase/release"
 release_bin="$GOPATH/bin/release"
 
+dryrun=""
+if [ $DRY_RUN == 'true' ]; then
+  dryrun="--dry-run"
+fi
+
 if [ -n "$RELEASE_TO_PROMOTE" ]; then
-  "$release_bin" promote-a-release --release="$RELEASE_TO_PROMOTE" --bucket-name="$BUCKET_NAME" --platform="$PLATFORM"
+  "$release_bin" promote-a-release --release="$RELEASE_TO_PROMOTE" --bucket-name="$BUCKET_NAME" --platform="$PLATFORM" $dryrun
   "$dir/send.sh" "Promoted $PLATFORM release $RELEASE_TO_PROMOTE ($BUCKET_NAME)"
 else
   "$release_bin" promote-releases --bucket-name="$BUCKET_NAME" --platform="$PLATFORM"
```

Comment:
Now that there's an expectation `DRY_RUN` is understood by this script, `promote-releases` should use dry run. Or at least crash.

Reply:
release's `promote-releases` doesn't actually support dryrun, only `promote-a-release` does.  I know pretty much every command could use it, just thought I'd stick to the one I was touching in this PR to start with.

It's fine for `promote-releases` to not support Windows because the Windows builder's doing that job of managing the admin stream JSON itself at the moment.

Reply:
@mlsteele Added an error for the combination of `--dry-run` and `promote-releases`, thanks for the suggestion!